foxpuppet/expected.py

```diff
@@ -0,0 +1,19 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+
+class new_window_is_opened(object):
+    """
+    An expectation for checking that a new window is found after requesting
+    it to be opened
+
+    :returns handle: The new window handle
+    """
+
+    def __init__(self, selenium, handles_before):
+        self.selenium = selenium
+        self.handles_before = handles_before
+
+    def __call__(self, *args, **kwargs):
+        return set(self.selenium.window_handles) - set(self.handles_before)
```

Comment:
This is returning a list. We should guard this with an if statement for when there is only one item in the list, and return a window object instead.